script.js

```diff
@@ -1,4 +1,107 @@
+// Global variables for state
+var numberOfPlayers = 2;
+var firstPlayer = "";
+var secondPlayer = "";
+var gameStage = "how many players";
+var playerTurn = 1;
+
+var dice1 = "";
+var dice2 = "";
+var firstPlayerNumber = 0;
+var secondPlayerNumber = 0;
+var firstPlayerCount = 0;
+var secondPlayerCount = 0;
+
+// Random dice roll mechanics
+var diceRoll = function () {
+  var randomNumber = Math.random() * 6;
+  var outputNumber = Math.floor(randomNumber) + 1;
+  return String(outputNumber);
+};
+
+// Dice roll stage
+var stageDiceRoll = function () {
+  dice1 = diceRoll();
+  dice2 = diceRoll();
+  var outputReply = `Your dice rolls are as follows: ${dice1}, ${dice2}. Please select which dice to be used first.`;
+  gameStage = "choose order";
+  return outputReply;
+};
+
+// Decide dice order
+var stageDiceOrder = function (userInput) {
+  if (userInput == 1) {
+    combinedNumber = dice1 + dice2;
+  }
+  if (userInput == 2) {
+    combinedNumber = dice2 + dice1;
+  }
+  if (playerTurn == 1 && gameStage == "choose order") {
+    firstPlayerNumber = combinedNumber;
+    var outputReply1 = `Your number is ${firstPlayerNumber}. ${secondPlayer}'s turn to roll and choose a dice order!`;
+  }
+  if (playerTurn == 2 && gameStage == "choose order") {
+    secondPlayerNumber = combinedNumber;
+    outputReply1 = `Your number is ${secondPlayerNumber}. ${firstPlayer}'s turn to roll and choose a dice order!`;
+  }
+  return outputReply1;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (gameStage == "how many players") {
+    numberOfPlayers = input;
+    gameStage = "name";
+    var myOutputValue = `Please key in name of 1st player.`;
+    return myOutputValue;
```

Comment:
you can just return the one line that you want to output in the return line.
so instead of what you wrote, you can do 

return `Please key in name of the 1st player`; 

so that you don't have to assign the string to a variable and then returning the  variable